### PR TITLE
feat(layout): replace taitank layout and remove layout directory

### PR DESCRIPTION
**Pull request by ilikethese**
# Pre-PR Checklist

- [x] I added/updated relevant documentation.
- [x] I followed the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters to submit commit message.
- [x] I squashed the repeated code commits.
- [x] I signed the [CLA].
- [x] I added/updated test cases to check the change I am making.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: dom/include/dom/taitank_layout_node.h [hunk 1]
```diff
@@ -22,16 +22,18 @@
 
 #include <cstdint>
 #include "dom/layout_node.h"
```
Comment (churchill-zhang) [inline dom/include/dom/taitank_layout_node.h:24]: 看起来没有中间层，如果使用yoga，这里是不是显式依赖taitank有问题？

**Review by ozonelmy — APPROVED** [review]
LGTM

**Review by zoomchan-cxj — APPROVED** [review]

**Review by lavnFan — DISMISSED** [review]

File: dom/CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +25,7 @@ project("dom")
 get_filename_component(PROJECT_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/.." REALPATH)
 
 include("${PROJECT_ROOT_DIR}/buildconfig/cmake/InfraPackagesModule.cmake")
+include("${PROJECT_ROOT_DIR}/buildconfig/cmake/GlobalPackagesModule.cmake")
```
Comment (medns) [inline dom/CMakeLists.txt:28]: unnessary include

File: dom/CMakeLists.txt [hunk 3]
```diff
@@ -53,8 +54,11 @@ if ("${LAYOUT_ENGINE}" STREQUAL "Yoga")
   target_compile_definitions(${PROJECT_NAME} PRIVATE "USE_YOGA")
   target_link_libraries(${PROJECT_NAME} PRIVATE yogacore)
 elseif ("${LAYOUT_ENGINE}" STREQUAL "Taitank")
-  target_compile_definitions(${PROJECT_NAME} PRIVATE "USE_TAITANK")
-  add_subdirectory(${PROJECT_ROOT_DIR}/layout ${CMAKE_CURRENT_BINARY_DIR}/taitank)
+  InfraPackage_Add(taitank
+      REMOTE "dom/third_party/taitank/1.0.2/git-repo.tgz"
+      LOCAL "third_party/taitank"
+      )
+  target_include_directories(${PROJECT_NAME} PUBLIC ${taitank_SOURCE_DIR}/src)
```
Comment (medns) [inline dom/CMakeLists.txt:61]: maybe expose from taitank project?

File: framework/examples/ios-demo/clean_and_generate.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (medns) [inline framework/examples/ios-demo/clean_and_generate.sh:1]: Looks like this file is not necessary?

**Review by medns — CHANGES_REQUESTED** [review]

File: dom/CMakeLists.txt (hunk 2 above)
Comment (ilikethese) [reply dom/CMakeLists.txt:28]: fix

File: dom/CMakeLists.txt (hunk 3 above)
Comment (ilikethese) [reply dom/CMakeLists.txt:61]: fix

File: framework/examples/ios-demo/clean_and_generate.sh (hunk 4 above)
Comment (ozonelmy) [reply framework/examples/ios-demo/clean_and_generate.sh]: fixed

**Review by medns — APPROVED** [review]
LGTM

**Review by ozonelmy — APPROVED** [review]

**Review by churchill-zhang — APPROVED** [review]

**Review by zoomchan-cxj — APPROVED** [review]

**Review by lavnFan — APPROVED** [review]

**Pull request merged**